lcls-twincat-pmps/PMPS/MajorComponents/PhotonEnergyInterface/FB_UndulatorSegment.TcPOU

```diff
@@ -3,8 +3,9 @@
   <POU Name="FB_UndulatorSegment" Id="{986484f4-a7ca-4f8c-bada-b360e7740d21}" SpecialFunc="None">
     <Declaration><![CDATA[FUNCTION_BLOCK FB_UndulatorSegment
 VAR_INPUT
-    (* Undulator period in millimeters, to be set by subclasses *)
+    (* Undulator period in millimeters, to be set by subclasses for HXR, read by pv for SXR, left here for backward compatability *)
```

Comment:
How does the code here know whether we're handling KFE vs LFE?

Reply:
It doesn't. There is another function block for L line that has a constant and pass it to this FB as input. For the K line the SXU function block assigns the pv value to this variable.